daemon/cmd/daemon.go

```diff
@@ -415,7 +415,18 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 
 	// Check the kernel if we can make use of managed neighbor entries which
 	// simplifies and fully 'offloads' L2 resolution handling to the kernel.
-	probeManagedNeighborSupport()
+	if !option.Config.DryMode {
+		err = probes.HaveManagedNeighbors()
+		switch {
+		case errors.Is(err, probes.ErrNotSupported):
+			log.Info("Managed Neigbors are not supported by the kernel")
+		case err != nil:
+			log.WithError(err).Warn("Unable to probe managed neighbor support")
```

Comment:
`probes.HaveProgramHelper` makes probe failures fatal, I think we should keep the same principle here. I find warnings confusing, as they're actually errors that fly under the radar. Does this require user intervention?

Reply:
Felt like a harsh intervention to fatal the agent for a failed probe that just determines how something is handled internally in the kernel (or does this enable any special features?), but I see that just warning about it doesn't really improve anything either. If we want to fatal I can move the error checking to the wrapper function `HaveManagedNeighbors` and fatal there as with the other probe APIs and simplify the error handling for the caller.

Reply:
I agree that we should keep the same principal, to keep things consistent. If a probe fails inconclusively something weird is going on.